imagebuildah/build.go

```diff
@@ -1812,25 +1812,13 @@ func (b *Executor) deleteSuccessfulIntermediateCtrs() error {
 }
 
 func (s *StageExecutor) EnsureContainerPath(path string) error {
-	// Check the path to see if it's a symlink, if
-	// so convert it to it's target for processing.
 	targetPath := filepath.Join(s.mountPoint, path)
-	fileDest, _ := os.Lstat(targetPath)
-	if fileDest != nil {
-		if fileDest.Mode()&os.ModeSymlink != 0 {
-			if symLink, err := resolveSymlink(s.mountPoint, path); err == nil {
-				targetPath = symLink
-			} else {
-				return errors.Wrapf(err, "error reading symbolic link to %q", targetPath)
-			}
-		}
-	}
-	_, err := os.Stat(targetPath)
+	_, err := os.Lstat(targetPath)
 	if err != nil && os.IsNotExist(err) {
 		err = os.MkdirAll(targetPath, 0755)
 	}
 	if err != nil {
-		return errors.Wrapf(err, "error ensuring container path %q", targetPath)
+		return errors.Wrapf(err, "error ensuring container path %q", path)
```

Comment:
I've reverted this function primarily back to the way Nalin had it two submits ago at: https://github.com/containers/buildah/commit/bc53b5d980005fe91329d47542fc2ac5769066a7#diff-65919b07714c5faa770fc0724c4ca886.

The only change I made to it was changing `os.Stat` to `os.Lstat()` and declaring targetPath rather than doing `filepath.Join()` in a couple of places.

Reply:
Not a great link, this is better:  https://github.com/containers/buildah/blob/bc53b5d980005fe91329d47542fc2ac5769066a7/imagebuildah/build.go#L1801

Reply:
The net change to `EnsureContainerPath()` since 34e7eba408282e890e61395b6d97e58b88e14d25 matches your description, and looks reasonable.  If it fixes #1537, then it LGTM.